test/integration/narrow/app/fuzzySearchService.test.js

```diff
@@ -174,4 +175,36 @@ describe('fuzzySearchService tests with filterBy', () => {
     expect(filteredResultPage2.count).toBe(11)
     expect(filteredResultPage2.rows.length).toBe(5)
   })
+
+  test('GET /paymentdata filters by single amount', async () => {
+    const amounts = ['5000-9999']
+    const filteredResult = await getPaymentData({ ...searchCriteria, filterBy: { amounts } })
+
+    filteredResult.rows.forEach(x => {
+      const matchingSet = paymentestdata.find(td => td.payee_name === x.payee_name)
+      expect(isInRange(matchingSet.total_amount, amounts[0])).toBe(true)
+    })
+  })
+
+  test('GET /paymentdata filters by multiple amounts', async () => {
+    const amounts = ['0-4999', '5000-9999']
+    const filteredResult = await getPaymentData({ ...searchCriteria, filterBy: { amounts } })
+
+    filteredResult.rows.forEach(x => {
+      const matchingSet = paymentestdata.find(td => td.payee_name === x.payee_name)
+      expect(amounts.some(x => isInRange(matchingSet.total_amount, x))).toBeTruthy()
```

Comment:
can we rename this x variable repeated twice to different one? more appropriate like r - range, will improve the readability

Reply:
Sorry where else is x defined? I am using x here because we know we are iterating through filteredResult, but I can't see x in the outer scope which might cause confusion.

Reply:
I noticed these 2 lines uses lambda parameter as x, which could be used with two different variable x, r.
though it will will , scope conflict may occur, and improves the readability.

filteredResult.rows.forEach(x => {
filteredResult.rows.forEach(x => {